### PR TITLE
doc updates

### PR DESCRIPTION
**Pull request by asonnenschein**
This MR includes updates to several different parts of the documentation:
- Updated the Installation/Quick Start README docs to include installation instructions from PyPI, and some links to docs about configuring virtual environments in Python.  Also updated the section on versions and stability in the README.
- Added clearer link to CONTRIBUTING.md from README.md.
- Updated CONTRIBUTING.md to align with changes to branch naming conventions described in https://github.com/planetlabs/planet-client-python/pull/1061.
- Updated Subscriptions CLI documentation to point to official Subscriptions API statuses documentation (https://github.com/planetlabs/planet-client-python/issues/1001).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,57 +14,54 @@ Version 2.0 includes support for the core workflows of the following APIs:
...
 Python](https://github.com/planetlabs/planet-client-python/projects/2),
-a complete rewrite and upgrade from the original [Planet Python 
-Client](https://developers.planet.com/docs/pythonclient/). If you 
-are looking for the source code to that library see the 
+a complete rewrite and upgrade from the original [Planet Python
+Client](https://developers.planet.com/docs/pythonclient/). If you
+are looking for the source code to that library see the
 [v1](https://github.com/planetlabs/planet-client-python/tree/v1) branch.
 
 The Planet SDK for Python is in 'pre-release' stages, working towards a solid
```
Comment (tbarsballe) [inline README.md:31]: While we're updating the README, this is no longer accurate. Could reword to just point to releases?

**Review by tbarsballe — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (asonnenschein) [reply README.md]: Yup!  The entire versions/stability section has been rewritten.

**Comment by asonnenschein** [comment]
Update - I also went ahead and updated `setup.py` to reflect organizational changes.

**Review by tbarsballe — APPROVED** [review]
Looks good to me

**Pull request merged**